src/lpython/parser/tokenizer.re

```diff
@@ -565,102 +205,37 @@ int Tokenizer::lex(Allocator &al, YYSTYPE &yylval, Location &loc, diag::Diagnost
-
-            // This is needed to ensure that 2.op.3 gets tokenized as
-            // TK_INTEGER(2), TK_DEFOP(.op.), TK_INTEGER(3), and not
-            // TK_REAL(2.), TK_NAME(op), TK_REAL(.3). The `.op.` can be a
-            // built-in or custom defined operator, such as: `.eq.`, `.not.`,
-            // or `.custom.`.
-            integer / defop {
-                lex_int_large(al, tok, cur,
-                    yylval.int_suffix.int_n,
-                    yylval.int_suffix.int_kind);
-                RET(TK_INTEGER)
-            }
-
+            'None' { RET(TK_NONE) }
 
             real { token(yylval.string); RET(TK_REAL) }
-            integer / (whitespace name) {
-                if (last_token == yytokentype::TK_NEWLINE) {
-                    uint64_t u;
-                    if (lex_int(tok, cur, u, yylval.int_suffix.int_kind)) {
-                            yylval.n = u;
-                            if (enddo_label_stack[enddo_label_stack.size()-1] == u) {
-                                while (enddo_label_stack[enddo_label_stack.size()-1] == u) {
-                                    enddo_label_stack.pop_back();
-                                    enddo_insert_count++;
-                                }
-                                enddo_newline_process = true;
-                            } else {
-                                enddo_newline_process = false;
-                            }
-                            RET(TK_LABEL)
-                    } else {
-                        token_loc(loc);
-                        std::string t = token();
-                        throw LFortran::parser_local::TokenizerError("Integer '" + t + "' too large",
-                            loc);
-                    }
-                } else {
-                    lex_int_large(al, tok, cur,
-                        yylval.int_suffix.int_n,
-                        yylval.int_suffix.int_kind);
-                    RET(TK_INTEGER)
-                }
-            }
-            integer {
-                lex_int_large(al, tok, cur,
-                    yylval.int_suffix.int_n,
-                    yylval.int_suffix.int_kind);
-                RET(TK_INTEGER)
-            }
-
-            [bB] '"' [01]+ '"' { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-            [bB] "'" [01]+ "'" { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-            [oO] '"' [0-7]+ '"' { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-            [oO] "'" [0-7]+ "'" { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-            [zZ] '"' [0-9a-fA-F]+ '"' { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-            [zZ] "'" [0-9a-fA-F]+ "'" { token(yylval.string); RET(TK_BOZ_CONSTANT) }
-
-            "&" ws_comment+ whitespace? "&"? {
-                line_num++; cur_line=cur; continue;
-            }
+            integer { token(yylval.string);  RET(TK_INTEGER) }
+            imag_number { token(yylval.string);  RET(TK_IMAG_NUM) }
```

Comment:
This returns integers and reals as strings. Rather, we should immediately convert integers, as before, just remove the support for Fortran specifix `_dp` style suffix. That is for performance reasons: we have to convert them anyway at some point, so if we do it right away, we are never even constructing the string representation of them.

Reply:
Perfect, I will do it.

Reply:
Awesome. Ping me once you do and I'll review again.